cli/src/bin/phylum.rs

```diff
@@ -58,16 +58,6 @@ async fn handle_commands() -> CommandResult {
 
     let matches = app.get_matches();
 
-    #[cfg(feature = "selfmanage")]
-    if let Some(matches) = matches.subcommand_matches("uninstall") {
-        return handle_uninstall(matches);
-    }
-
-    #[cfg(feature = "extensions")]
-    if let Some(matches) = matches.subcommand_matches("extension") {
-        return handle_extensions(matches).await;
-    }
-
```

Comment:
These commands were at this point in the function to avoid requiring authentication to run them.

Reply:
It took a few hours for this to finally hit me... But now I see that you've solved this with the `async` block around `PhylumApi::new()`...

Reply:
Authentication will not happen until the `api` future is polled; this way, only the functions calling `api.await?` (see [this](https://github.com/phylum-dev/cli/pull/415/files#diff-ad5eecad8cf0e108661eb7654555c2e7b0fb3aa2f4b0d01549853615fd27dd34R182) for an example) will actually trigger the authentication process. I feel like this is a better pattern to use in general, and we could further refactor this module to that end.